public/app/features/plugins/extensions/getPluginExtensions.test.tsx

```diff
@@ -564,3 +575,178 @@ describe('getPluginExtensions()', () => {
     );
   });
 });
+
+describe('getObservablePluginExtensions()', () => {
+  const extensionPointId = 'grafana/dashboard/panel/menu/v1';
+  const pluginId = 'grafana-basic-app';
+
+  beforeEach(() => {
+    pluginExtensionRegistries.addedLinksRegistry = new AddedLinksRegistry();
+    pluginExtensionRegistries.addedComponentsRegistry = new AddedComponentsRegistry();
+    pluginExtensionRegistries.addedLinksRegistry.register({
+      pluginId,
+      configs: [
+        {
+          title: 'Link 1',
+          description: 'Link 1 description',
+          path: `/a/${pluginId}/declare-incident`,
+          targets: extensionPointId,
+          configure: jest.fn().mockReturnValue({}),
+        },
+      ],
+    });
+
+    pluginExtensionRegistries.addedComponentsRegistry.register({
+      pluginId,
+      configs: [
+        {
+          title: 'Component 1',
+          description: 'Component 1 description',
+          targets: extensionPointId,
+          component: () => {
+            return <div>Hello world!</div>;
+          },
+        },
+      ],
+    });
+  });
+
+  it('should emit the initial state when no changes are made to the registries', () => {
+    const observable = getObservablePluginExtensions({ extensionPointId });
+
+    observable.subscribe((value: ReturnType<GetExtensions>) => {
+      expect(value.extensions).toHaveLength(2);
+      expect(value.extensions[0].pluginId).toBe(pluginId);
+      expect(value.extensions[1].pluginId).toBe(pluginId);
+    });
+  });
```

Comment:
You might want to replace this with the `toEmitValuesWith` matcher as it has better handling of RxJs edge cases
```suggestion
it('should emit the initial state when no changes are made to the registries', async () => {
    const observable = getObservablePluginExtensions({ extensionPointId }).pipe(first());

    await expect(observable).toEmitValuesWith((received) => {
      const { extensions } = received[0];
      expect(extensions).toHaveLength(2);
      expect(extensions[0].pluginId).toBe(pluginId);
      expect(extensions[1].pluginId).toBe(pluginId);
    });
  });
```